entityDumper/configurationDumper.go

```diff
@@ -0,0 +1,123 @@
+package entityDumper
+
+import (
+	"bufio"
+	"database/sql"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/uyuni-project/inter-server-sync/dumper"
+	"github.com/uyuni-project/inter-server-sync/dumper/packageDumper"
+	"github.com/uyuni-project/inter-server-sync/schemareader"
+	"github.com/uyuni-project/inter-server-sync/sqlUtil"
+	"os"
+)
+
+func ConfigTableNames() []string {
+	return []string{
+		"rhnconfigfile",
+		"rhnconfigfilename",
+		"rhnconfigrevision",
+		"rhnconfigcontent",
+		"rhnconfigchannel",
+		"rhnconfigfilestate",
+		"rhnregtokenconfigchannels",
+		"rhnserverconfigchannel",
+		"rhnsnapshotconfigchannel",
+		"susestaterevisionconfigchannel",
+		"rhnconfiginfo",
+		"rhnconfigfilefailure",
+		"rhnchecksum",
+		"rhnchecksumtype",
+		"web_contact",
+	}
+}
+
+func DumpConfigs(options ChannelDumperOptions) {
+	var outputFolderAbs = options.GetOutputFolderAbsPath()
+	db := schemareader.GetDBconnection(options.ServerConfig)
+	defer db.Close()
+	//file, err := os.OpenFile(outputFolderAbs + "/sql_statements.sql", os.O_APPEND | os.O_WRONLY | os.O_CREATE, 0600)
+	file, err := os.Create(outputFolderAbs + "/configurations.sql")
+	if err != nil {
+		log.Fatal().Err(err).Msg("error creating sql file")
+		panic(err)
+	}
+	defer file.Close()
+	bufferWriter := bufio.NewWriter(file)
+	defer bufferWriter.Flush()
+
+	bufferWriter.WriteString("BEGIN;\n")
+	processConfigs(db, bufferWriter, loadConfigsToProcess(db, options), options)
+
+	bufferWriter.WriteString("COMMIT;\n")
+}
+
+func loadConfigsToProcess(db *sql.DB, options ChannelDumperOptions) []string {
+	labels := channelsProcess{make(map[string]bool), make([]string, 0)}
+	for _, singleChannel := range options.ConfigLabels {
+		if _, ok := labels.channelsMap[singleChannel]; !ok {
+			labels.addChannelLabel(singleChannel)
+		}
+	}
+
+	for _, channelChildren := range options.ChannelWithChildrenLabels {
+		if _, ok := labels.channelsMap[channelChildren]; !ok {
+			labels.addChannelLabel(channelChildren)
+			childrenChannels := sqlUtil.ExecuteQueryWithResults(db, childChannelSql, channelChildren)
+			for _, cChannel := range childrenChannels {
+				cLabel := fmt.Sprintf("%v", cChannel[0].Value)
+				if _, okC := labels.channelsMap[cLabel]; !okC {
+					labels.addChannelLabel(cLabel)
+				}
+			}
+
+		}
+	}
+	return labels.channels
+}
+
+func processConfigs(db *sql.DB, writer *bufio.Writer, labels []string, options ChannelDumperOptions) {
+	log.Info().Msg(fmt.Sprintf("%d channels to process", len(labels)))
+	schemaMetadata := schemareader.ReadTablesSchema(db, ConfigTableNames())
+	log.Debug().Msg("channel schema metadata loaded")
+	configLabels, err := os.Create(options.GetOutputFolderAbsPath() + "/exportedConfigs.txt")
+	if err != nil {
+		log.Fatal().Err(err).Msg("error creating exportedConfigChannel file")
+		panic(err)
```

Comment:
we don't need the panic, I think the fatal will call it

Reply:
@rjmateus that's a really interesting point. The Fatal calls `os.Exit(1)` which just kills the program immediately without calling defers, so effectively leaving the connections open. However there is log.Panic() method. So I guess this is what we want. Great point.